### PR TITLE
feat(fe,meta): support logical view

### LINKED ISSUE #2317: Support logical view (CREATE VIEW commands)
**Is your feature request related to a problem? Please describe.**

Currently, RisingWave supports `CREATE MATERIALIZED VIEW` but not `CREATE VIEW`. It will be easier to generate some similar tests between logical view and batch view if we support it.

**Describe the solution you'd like**

Support `CREATE VIEW`.

**Describe alternatives you've considered**

As a first-class mview database, `CREATE VIEW` may be ambiguous, `CREATE LOGICAL VIEW` is also OK.

**Additional context**

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

close #2317

See `e2e_test/batch/basic/logical_view.slt.part` for examples

idea: 
- store sql (see `View` in `catalog.proto`).
- create & drop are straightforward
- query: get view from catalog, and bind it as subquery relation (see `binder/relation/table_or_source.rs`)

limitations:
- View's dependants not resolved. e.g., when view `V` depends on mv (or table) `MV`, we can't drop `MV` before dropping `V`. But if mv (or view) `MV` depends on `V`, we can drop `V`. 
  
  The difficulty is that currently after binding, view is completely replaced by the sql and the information of the view is lost. To support it, we may collect the dependent view ids in binder. Note that currently for mv, dependent relation ids are resolved in meta from stream fragment plan. For view, they are resolved in frontend from batch plan.

- How to handle * in case of ddl (specifically adding column) is not yet considered

  example: `create table t(x int); create view v as select * from t; alter table t add column y int;` In PG, `v` only outputs `x` without `y`. This means `*` is resolved before the view is stored. This behavior is expected also because we can specify column names when creating views. It only makes sense if the view outputs fixed number of columns.

  The difficulty is that `*` is resolved in binder, but after binding, we can't restore the `BoundQuery` to AST and unparse it. One solution is to serializing plan instead of sql, but as discussed earlier we don't want to persist plan because it may be subject to change. Another solution is to rewriting AST while binding, but it may be a little bit hacky.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

* SQL commands

### Release note

Support `CREATE VIEW`

See `e2e_test/batch/basic/logical_view.slt.part` for examples

### CODE REVIEW

**Comment by xxchan** [comment]
Not work yet
```sql
dev=> create view v as select 1;
CREATE_VIEW
dev=> select * from v;
--
(1 row)

dev=> create table t(x int);
CREATE_TABLE
dev=> insert into t values (1),(2);
INSERT 0 2
dev=> create view vv as select * from t;
CREATE_VIEW
dev=> select * from vv;
--
(2 rows)
```

Other outstanding tasks:
- [ ] drop view
- [ ] resolve dependent tables
- [ ] consider how to handle `*` in case of ddl

**Review by HuaHuaY — APPROVED** [review]
LGTM. Good Job!

File: src/frontend/src/handler/create_index.rs [hunk 1]
```diff
@@ -346,26 +324,7 @@ pub async fn handle_create_index(
...
-            let index_name = Binder::resolve_index_name(name.clone())?;
-
-            let catalog_reader = session.env().catalog_reader().read_guard();
-            let (_, schema_name) =
-                catalog_reader.get_table_by_name(db_name, schema_path, &table_name)?;
-
-            if let Err(e) =
-                catalog_reader.check_relation_name_duplicated(db_name, schema_name, &index_name)
-            {
+            if let Err(e) = session.check_relation_name_duplicated(index_name.clone()) {
```
[the remark this answers is not in the document]
Comment (HuaHuaY) [reply src/frontend/src/handler/create_index.rs:327]: `resolve_index_name` in function `gen_create_index_plan` will check this. It's my mistake. There is no error here.

File: src/frontend/src/observer/observer_manager.rs [hunk 2]
```diff
@@ -86,16 +87,16 @@ impl ObserverState for FrontendObserverNode {
         match resp.info {
             Some(Info::Snapshot(snapshot)) => {
                 for db in snapshot.databases {
-                    catalog_guard.create_database(db)
```
Comment (yezizp2012) [inline src/frontend/src/observer/observer_manager.rs:89]: We should also create all views when handling initialization notification. The failure in deterministic batch recovery test is possibly cause by it:
```
deterministic simulation e2e, ci-3cn-1fe, recovery, batch
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_US.utf8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
thread '<unnamed>' panicked at 'failed to run test after retry 5 times: test error at e2e_test/batch/./basic/logical_view.slt.part:7: query failed: db error: ERROR: QueryError: Catalog error: table or source not found: v1
```

File: src/frontend/src/handler/create_sink.rs [hunk 3]
```diff
@@ -60,58 +55,26 @@ pub fn gen_sink_plan(
...
-            schema.id(),
-            associated_table_catalog.id().table_id,
-            associated_table_catalog.table_desc(),
-        )
-    };
+    let (database_id, schema_id, associated_table_catalog) =
+        session.get_table_catalog_for_create(schema_name, &associated_table_name)?;
 
-    let sink_name = Binder::resolve_sink_name(stmt.sink_name)?;
+    let (_, sink_name) = Binder::resolve_schema_qualified_name(db_name, stmt.sink_name)?;
```
[the remark this answers is not in the document]
Comment (xxchan) [reply src/frontend/src/handler/create_sink.rs:63]: My mistake here

File: src/frontend/src/observer/observer_manager.rs (hunk 2 above)
Comment (xxchan) [reply src/frontend/src/observer/observer_manager.rs:89]: (Why GitHub is green when the test failed..

File: proto/user.proto [hunk 4]
```diff
@@ -57,6 +57,7 @@ message GrantPrivilege {
     uint32 source_id = 4;
     uint32 all_tables_schema_id = 5;
     uint32 all_sources_schema_id = 6;
+    uint32 view_id = 8;
```
Comment (yezizp2012) [inline proto/user.proto:60]: We might need to support grant/revoke privileges for views (and all views in schema maybe) in user service. But that's non-related to this PR. Cc @HuaHuaY

**Review by yezizp2012 — APPROVED** [review]
generally LGTM, great job!!!

**Pull request merged**